### ISSUE TITLE
Merge bitbucket cloud SCM generator from previous applicationset repo

### ISSUE THREAD

**Issue by jdomag** [argoproj/argo-cd#10143]
The bitbucket cloud SCM provider was added [in this PR](https://github.com/argoproj/applicationset/pull/556/files)
Unfortunately that PR was not included when the applicationset project has been merged into arogcd. 
Would be great to have this feature.

**Comment by jdomag** [comment]
@wtam2018 
You took a part in PR approval process, maybe you could help to prioritize this feature? It's a shame that ready to go / working feature is not used.

**Comment by jdomag** [comment]
@jessesuen can you label that one with component:applications-set, please?
/label component:applications-set

**Comment by mmerrill3** [comment]
Hi @jdomag , if there's anything you need from me, just let me know.

**Comment by jdomag** [comment]
> Hi @jdomag , if there's anything you need from me, just let me know.

@mmerrill3 will you be able to merge your PR from previous repo to main argocd repository? Would be great to use bitbucket in a generator.

**Comment by mmerrill3** [comment]
@jdomag, sure, I can do that, I'll open a PR this weekend sometime.

**Comment by mmerrill3** [comment]
Hi @jdomag, I just had time to check this, and I see the PR in the master branch of argo-cd for bitbucket-cloud.  Is there a particular part of the PR that's missing?

**Comment by jdomag** [comment]
@mmerrill3 
I can't see scm provider of bitbucket in [scm_provider.go ](https://github.com/argoproj/argo-cd/blob/master/applicationset/generators/scm_provider.go) - there is only Bitbucket Server.

**Comment by mmerrill3** [comment]
@jdomag yeah, I see, looks like the new files are there, and the updated files are not.
